Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,135 @@
+* The second egress section allows ``mediabot`` pods to access ``kube-dns`` service. Note that ``rules: dns`` instructs Cilium to inspect and allow DNS lookups matching specified patterns. In this case, inspect and allow all DNS queries.  
+ 
+Note that with this policy the ``mediabot`` doesn't have access to any internal cluster service other than ``kube-dns``. Refer :ref:`Network Policy` to learn more about policies for controlling access to internal cluster services.
+
+Let's apply the policy:
+
+.. parsed-literal::
+
+  $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes-dns/dns-matchname.yaml
+
+Testing the policy, we see that ``mediabot`` has access to ``api.twitter.com`` but doesn't have access to any other external service, for e.g., ``help.twitter.com``. 
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+    ...
+
+    $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+    ^C  
+
+Step 4: DNS Policies Using Patterns 
+===================================
+
+The above policy controlled DNS access based on exact match of the DNS domain name. Often, it is required to allow access to a subset of domains. Let's say, in the above example, ``mediabot`` pods need access to any Twitter sub-domain, for e.g., the pattern ``*.twitter.com``. We can achieve this easily by changing the ``toFQDN`` rule to use ``matchPattern`` instead of ``matchName``.
+
+.. literalinclude:: ../../examples/kubernetes-dns/dns-pattern.yaml
+
+.. parsed-literal::
+
+  $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-dns/dns-pattern.yaml
+
+Test that ``mediabot`` has access to multiple Twitter services for which the DNS matches the pattern ``*.twitter.com``. It is important to note and test that this doesn't allow access to ``twitter.com`` because the ``*.`` in the pattern requires one subdomain to be present in the DNS name. You can simply add more ``matchName`` and ``matchPattern`` clauses to extend the access. 
+(See :ref:`DNS based` policies to learn more about specifying DNS rules using patterns and names.)
+
+.. parsed-literal:: 
+  
+   $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+   ...
+
+   $ kubectl exec -it mediabot -- curl -sL https://about.twitter.com
+   ... 
+
+   $ kubectl exec -it mediabot -- curl -sL https://twitter.com
+   ^C 
+ 
+Step 5: Combining DNS, Port and L7 Rules
+========================================
+
+The DNS-based policies can be combined with port (L4) and API (L7) rules to further restrict the access. In our example, we will restrict ``mediabot`` pods to:
+
+* Access Twitter services only on ports ``443`` and port ``80``. The ``toPorts`` section in the policy below achieves the port-based restrictions.
+* On port ``80``, restrict access to only perform ``HTTP GET``. The ``rules: http`` section in the policy below achieves the L7 restrictions for port ``80``.
```

Comment:
What's the practical purpose of this second example? It isn't really logical to restrict one port more than the other.

Reply:
My intention was to give an example of combining DNS with L7 along with the port. Sometimes a service may expose multiple ports for eg. for admin user vs general users, for exposing over internet vs for internal access, for metrics vs other resources, ... these ports can have different network security profiles. But my intention is to give an example of L7 combination with DNS. Open to either removing the L, just mentioning L7 in text or using some other example.

Reply:
I don't like including examples which give a wrong impression. DNS+L7 to external services will typically not work due to TLS for now. It's also very tempting to just add the http section to the port 443 and that will not work. It will be unclear to the user without further references why that doesn't work. If you want to include L7, find a real example to a service that is typically unencrypted.

Reply:
Unencrypted public service would be hard. Let me just put the wording and reference to L7. I will limit this to l4 example